### DISCUSSION TITLE
Can use rustls instead of OpenSSL?

### DISCUSSION THREAD

**Discussion by kyoheiu** [flo-at/minmon#60] [General]
Thank you for this amazing project.
I wonder why openssl is needed: Is this because lettre's feature?
FWIW it seems lettre supports rustls: https://docs.rs/lettre/latest/lettre/#smtp-over-tls-via-the-rustls-crate

**Comment by flo-at** [comment]
Thanks for the idea! I chose native-tls (which leads to OpenSSL on Linux) because of the fact that it's a security critical library that can be updated without updating MinMon thanks to dynamic linking. Also with rustls the application has to ship all the root certificates if I'm not mistaken here.

But it should be possible to add a compile-time option to MinMon so the user can choose himself. Let me double check whether all dependencies support it, then I can add an option to Cargo.toml.

**Reply by kyoheiu** [reply]
That design decision makes sense.
And would be great to have the option to use rustls! Thank you.

**Comment by flo-at** [comment]
I'll lock the discussion in favor to the ticket.
